examples/helloworld/helloworld.cc

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <opentelemetry/trace/span_id.h>
+
+#include <iostream>
+
+int main() {
+  opentelemetry::trace::SpanId span_id;
+  std::cout << "Span ID: " << span_id.ToHex() << "\n";
```

Comment:
```suggestion
  std::cout << "Span ID: " << span_id.ToHex() << std::endl;
```

Reply:
How do people feel about endl vs "\n" in general? AFAIK "\n" works everywhere (even \r\n platforms) and endl forces a flush.

Reply:
Whatever works well for a customer C++ app. Internally in SDK I see how it would have been easier to create a set of printf-style macros, rerouting all internal SDK logs to whatever alternate facility (syslog, file, ETW, OutputDebugString) with a wrapper like:

```
#ifndef HAVE_LOGGING
    /* No logging */
#define LOG_DEBUG(fmt, ...)
#define LOG_TRACE(fmt, ...)
#define LOG_INFO(fmt,  ...)
#define LOG_WARN(fmt,  ...)
#define LOG_ERROR(fmt, ...)
#else
#ifdef NDEBUG
#define LOG_DEBUG(fmt_, ...)
#else
#define LOG_DEBUG(fmt_, ...)    OTELSDK_LOG_(DebugLevel::Detail,  getLogComponent(), fmt_, ##__VA_ARGS__)
#endif
#define LOG_TRACE(fmt_, ...)    OTELSDK_LOG_(DebugLevel::Detail,  getMATSDKLogComponent(), fmt_, ##__VA_ARGS__)
#define LOG_INFO(fmt_, ...)     OTELSDK_LOG_(DebugLevel::Info,    getMATSDKLogComponent(), fmt_, ##__VA_ARGS__)
#define LOG_WARN(fmt_, ...)     OTELSDK_LOG_(DebugLevel::Warning, getMATSDKLogComponent(), fmt_, ##__VA_ARGS__)
#define LOG_ERROR(fmt_, ...)    OTELSDK_LOG_(DebugLevel::Error,   getMATSDKLogComponent(), fmt_, ##__VA_ARGS__)
#endif
```

Then instrumenting code with printf-style macros. In those macros you don't need to care about the line endings because the underlying implementation will either add these or not, depending on where you internal SDK logs go.

Reply:
There's a core guideline for avoiding std::endl
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rio-endl

Reply:
I'm starting to like CppCore more.